pyccl/core.py

```diff
@@ -194,6 +194,23 @@ class Cosmology(object):
                                      "HMCode_logT_AGN": 7.8}}
 
     """
+    # Go through all functions in the main package and the subpackages
+    # and make every function that takes `cosmo` as its first argument
+    # an attribute of this class.
+    from . import background, boltzmann, bcm, \
+        cls, correlations, covariances, \
+        pk2d, power, tracers, halos, nl_pt
+    subs = [background, boltzmann, bcm, cls, correlations, covariances,
+            pk2d, power, tracers, halos, nl_pt]
+    funcs = [getmembers(sub, isfunction) for sub in subs]
+    funcs = [func for sub in funcs for func in sub]
+    for name, func in funcs:
+        pars = signature(func).parameters
+        if list(pars)[0] == "cosmo":
+            vars()[name] = func
```

Comment:
can you explain what this line is doing?

Reply:
For every function in the functions exposed in the `__init__.py` files, check the function signature. If the first argument of the funciton is "cosmo", then add it to the local (class) variables as an attribute with the same name.

Reply:
If it's not clear in the source code, I can always add comments! Let me know.

Reply:
Yeah, can you add a comment? Took me a bit to figure out what `vars()` (with no argument) did

Reply:
I actually just realised there is already a comment above the loop, explaining what I do in the lines below. Is this enough explanation?